tflearn/utils.py

```diff
@@ -384,3 +384,52 @@ def autoformat_padding(padding):
         return str.upper(padding)
     else:
         raise Exception("Unknow padding! Accepted values: 'same', 'valid'.")
+
+
+# Auto format filter size
+# Output shape: (rows, cols, input_depth, out_depth)
+def autoformat_filter_conv3d(fsize, in_depth, out_depth):
+    if type(fsize) is int:
+        return [fsize, fsize, fsize, in_depth, out_depth]
+    elif type(fsize) in [tuple, list]:
+        if len(fsize) == 3:
+            return [fsize[0], fsize[1],fsize[2], in_depth, out_depth]
+        else:
+            raise Exception("filter length error: " + str(len(fsize))
+                            + ", only a length of 3 is supported.")
+    else:
+        raise Exception("filter format error: " + str(type(fsize)))
+
+
+# Auto format stride for 3d convolution
+def autoformat_stride_3d(strides):
+    if type(strides) is int:
+        return [1, strides, strides, 1]
+    elif type(strides) in [tuple, list]:
```

Comment:
https://www.python.org/dev/peps/pep-0008/
Object type comparisons should always use isinstance() instead of comparing types directly.

Yes: if isinstance(obj, int):

No:  if type(obj) is type(1):